cmake/common.cmake

```diff
@@ -115,6 +116,17 @@ macro(spl_create_component)
     # Collect all component names for later usage (e.g., in an extension)
     list(APPEND COMPONENT_NAMES ${component_name})
     set(COMPONENT_NAMES ${COMPONENT_NAMES} PARENT_SCOPE)
+    
+    # Collect all component information for sphinx documentation
+    set(_component_info "{
+\"name\": \"${component_name}\",

```

Comment:
We use json strings because the content will be written in a config.json file during configure. Also, CMake supports manipulating JSON strings. See the string(JSON ...) documentation.